# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.cpp
# Commit: 22854b710fbf
# Full Hash: 22854b710fbfb779ac6b19fe4c9ac2570189cec0
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 4531d5c25694 -r 22854b710fbf js/src/frontend/Stencil.cpp
--- a/js/src/frontend/Stencil.cpp	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/src/frontend/Stencil.cpp	Fri Mar 05 19:10:07 2021 +0000
@@ -927,9 +927,10 @@
   if (!group) {
     return false;
   }
-  RootedShape shape(cx, EmptyShape::getInitialShape(
-                            cx, &JSFunction::class_, TaggedProto(proto),
-                            /* nfixed = */ 0, ObjectFlags()));
+  RootedShape shape(
+      cx, EmptyShape::getInitialShape(cx, &JSFunction::class_, cx->realm(),
+                                      TaggedProto(proto),
+                                      /* nfixed = */ 0, ObjectFlags()));
   if (!shape) {
     return false;
   }